LoopKit/Persistence/PersistenceController.swift

```diff
@@ -235,7 +235,7 @@ extension PersistenceController: CustomDebugStringConvertible {
 
 extension PersistenceController {
     func storeAnchor(_ anchor: HKQueryAnchor?, key: String) {
-        managedObjectContext.performAndWait {
+        managedObjectContext.perform {
```

Comment:
Reverting this change; I've seen failures with and without it.

Reply:
[nit] Do we want `perform` or `performAndWait` here now that you've found the cause of the failures?

Reply:
Yes, this is intentional; preferring perform, as it was originally, as not to block other work. I added the blocking/sync write to see if we were trying to read before it had been written, but the problem was actually that we were using the anchor before it was read from the cache.